### ISSUE TITLE
Setting 

### ISSUE THREAD

**Issue by kevlarr** [directus/directus#8569]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

The google auth library allows passing in either a path to a key file _or_  a [credentials object](https://github.com/googleapis/google-auth-library-nodejs/blob/main/src/auth/googleauth.ts#L74)  that has `private_key` and `client_email` properties.

As far as I can tell, it's not possible to pass these through with Directus. For instance, given how the [config is created](https://github.com/directus/directus/blob/main/api/src/utils/get-config-from-env.ts#L27) I would expect that I could use double-underscores in the following environment variables...

```
GOOGLE_STORAGE_CREDENTIALS__PRIVATE_KEY = "...key..."
GOOGLE_STORAGE_CREDENTIALS__CLIENT_EMAIL = "email@example.com"
```
... but by default anything split is camel-cased when [actually instantiated](https://github.com/directus/directus/blob/main/api/src/storage.ts#L30), meaning that I get a `credentials` object with `privateKey` and `clientEmail` properties, rather than `private_key` and `client_email` ones.

Is there something I'm doing wrong?

### To Reproduce

Instead of defining...
```
STORAGE_GOOGLE_KEY_FILENAME = "path/to/key/file.json
```
... pointing to the key file, define environment variables that have the actual values

```
STORAGE_GOOGLE_CREDENTIALS__PRIVATE_KEY = "...key..."
STORAGE_GOOGLE_CREDENTIALS__CLIENT_EMAIL = "...email..."
```

### What version of Directus are you using?

v9.0.0-rc.95

### What version of Node.js are you using?

16.2.0

### What database are you using?

Postgres

### What browser are you using?

Firefox

### What operating system are you using?

Ubuntu

### How are you deploying Directus?

Docker

**Comment by azrikahar** [comment]
Technically the `getConfigFromEnv` function does have the `type` param to use underscore instead of the default camelCase:

https://github.com/directus/directus/blob/be29a83c96222228dd5f13f5c1f152eff3fcf783/api/src/utils/get-config-from-env.ts#L5-L9

Which is used in Oauth configs (note the last argument, `'underscore'`):

https://github.com/directus/directus/blob/be29a83c96222228dd5f13f5c1f152eff3fcf783/api/src/grant.ts#L21

This is because the [grant](https://www.npmjs.com/package/grant) packages sets all keys in underscore format.

However in this Google Cloud Storage case (Nodejs SDK), all properties are in camel case _except_ `credentials.private_key` and `credentials.client_email`... which is pretty inconsistent imo... Reference: https://googleapis.dev/nodejs/storage/latest/global.html#StorageOptions

Not super sure if the added logic (ensure those two properties remain as underscore format in `@directus/drive-gcs`) to accommodate this quirk is okay or not versus the existing supported key file approach. @rijkvanzanten any thoughts on this?

**Comment by stx-chris** [comment]
Hi @rijkvanzanten this (ancient) fix seems to be somehow broken; When a `.env` assignment like this is performed:

`STORAGE_GCS_CREDENTIALS='{"client_email":"...@....iam.gserviceaccount.com","private_key":"-----BEGIN PRIVATE KEY-----\nMII...Hf8E=\n-----END PRIVATE KEY-----\n"}'`

as indicated in #7201 and #7257 , then it is resolved on bootup as follows and renders the credentials unusable:

![image](https://user-images.githubusercontent.com/62717406/191665391-f1d2408f-fbae-4947-8411-a355a1d1973a.png)

Maybe a recent regression?

**Comment by azrikahar** [comment]
@stx-chris seems like the json is being (mistakenly) auto casted to csv based on the commas, which is likely related to #12330. 

That being said, #12727 added a fix for this to support `json:` type cast prefix, can you try whether that works for you?

Issues with auto type casts in general are currently tracked in #9521 for v10.

**Comment by stx-chris** [comment]
Wow! Thanks @azrikahar, that fixed it. May I ask then to include the `json:` prefix description to the [docs?](https://docs.directus.io/self-hosted/config-options.html#environment-variable-files). Many thanks again!

**Comment by azrikahar** [comment]
@stx-chris yes it definitely should be added! Do you want to make a quick PR for it? 😊
